tests/com/magento/idea/magento2plugin/inspections/xml/PreferenceDeclarationInspectionTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.inspections.xml;
+
+import com.magento.idea.magento2plugin.magento.files.ModuleDiXml;
+
+public class PreferenceDeclarationInspectionTest extends InspectionXmlFixtureTestCase {
+
+    private static final String ARGUMENT_VALUE_IS_EMPTY =
+            "inspection.error.idAttributeCanNotBeEmpty";
+
+    private static final String CLASS_DOES_NOT_EXIST =
+            "inspection.warning.class.does.not.exist";
+
+    private static final String EXISTENT_CLASS_ONE =
+            "Magento\\Catalog\\Api\\ProductRepositoryInterface";
+
+    private static final String EXISTENT_CLASS_TWO =
+            "Foo\\Bar\\Model\\Logger";
+
+    private static final String NOT_EXISTENT_CLASS =
+            "Not\\Existent\\Class";
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        myFixture.enableInspections(PreferenceDeclarationInspection.class);
+    }
+
+    /**
+     * Test for an error for the "type" attribute because it is empty.
+     * <preference for="" type="Foo\Bar\Model\Logger"/>
+     */
+    public void testAttrArgForValuesIsEmpty() {
```

Comment:
@Iamwade, sorry, but I cannot understand this...you are testing type attribute, but method annotation, description, name and test data file name says that you are testing For attribute. Could you check it?

Reply:
Fixed.

Reply:
Thank you!